test/functional/rpc_rawtransaction.py

```diff
@@ -69,24 +74,129 @@ def setup_network(self):
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+        self.generate_and_sync(node=0, blocks=5)
+
+        self.getrawtransaction_tests()
+        self.createrawtransaction_tests()
+        self.sendrawtransaction_tests()
+        self.sendrawtransaction_testmempoolaccept_tests()
+        self.decoderawtransaction_tests()
+        self.transaction_version_number_tests()
+        if not self.options.descriptors:
+            self.raw_multisig_transaction_legacy_tests()
+
+    def getrawtransaction_tests(self):
+        addr = self.nodes[1].getnewaddress()
+        txid = self.nodes[0].sendtoaddress(addr, 10)
+        self.generate_and_sync(node=0, blocks=1)
+        vout = find_vout_for_address(self.nodes[1], txid, addr)
+        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
+        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
+        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
+        self.generate_and_sync(node=0, blocks=1)
+
+        # Make a tx by sending, then generate 2 blocks; block1 has the tx in it
+        tx = self.nodes[2].sendtoaddress(self.nodes[1].getnewaddress(), 1)
+        block1, block2 = self.nodes[2].generate(2)
+        self.sync_peers()
+
+        err_msg = (
+            "No such mempool transaction. Use -txindex or provide a block hash to enable"
+            " blockchain transaction queries. Use gettransaction for wallet transactions."
+        )
+
+        for n in [0, 3, 4, 5]:
+            self.log.info(
+                f"Test getrawtransaction {'with' if n == 0 or n == 5 else 'without'} -txindex,"
+                f" {'with' if n >= 4 else 'without'} -blocksonly"
+            )
+
+            # 1. valid parameters - supply txid along with various valid values for verbose
+            if n == 0 or n == 5:
+                # with -txindex
+                for verbose in [None, 0, False]:
+                    gottx = self.nodes[n].getrawtransaction(txId, verbose)
+                    assert_equal(gottx, rawTxSigned['hex'])
+                    assert 'in_active_chain' not in gottx
+                for verbose in [1, True]:
+                    # We only check the "hex" field of the output so we don't need to update
+                    # this test every time the output format changes.
+                    gottx = self.nodes[n].getrawtransaction(txId, verbose)['hex']
+                    assert_equal(gottx, rawTxSigned['hex'])
+                    assert 'in_active_chain' not in gottx
+            else:
+                # without -txindex
+                for verbose in [None, 0, False, 1, True]:
+                    assert_raises_rpc_error(-5, err_msg, self.nodes[n].getrawtransaction, txid, verbose)
```

Comment:
I would expect `txId` instead of `txid` here. Or can you explain this please?

Reply:
Good catch! This illustrates why using `txid` and `txId` variable names in the same test isn't ideal. Fixed in commit "Test src/validation::GetTransaction() with -blocksonly".